hazelcast/src/main/java/com/hazelcast/scheduledexecutor/impl/ScheduledExecutorServiceProxy.java

```diff
@@ -493,11 +503,10 @@ private String extractNameOrGenerateOne(Object command) {
         return createFutureProxy(uuid, taskName);
     }
 
-    @SuppressWarnings("unchecked")
     private <T> T initializeManagedContext(Object object) {
         ManagedContext context = getNodeEngine().getSerializationService().getManagedContext();
-        if (object instanceof NamedTaskDecorator) {
-            ((NamedTaskDecorator) object).initializeContext(context);
+        if (object instanceof AbstractTaskDecorator) {
```

Comment:
Same concerns as client proxy

Reply:
It seems to me that an object implementing NamedTask is handled by the `else` block. As it worked before.